src/eigsolve/arnoldi.jl

```diff
@@ -348,6 +348,12 @@ function _schursolve(A, x₀, howmany::Int, which::Selector, alg::Arnoldi)
     β = normres(fact)
     tol::eltype(β) = alg.tol
 
+    if eltype(fact) <: Real && krylovdim < 4
+        error("krylov dimension should be at least 4 to avoid getting stuck in the Arnoldi process in real arithmetic")
+    elseif krylovdim < 2
+        error("krylov dimension should be at least 2 to avoid getting stuck in the Arnoldi process in complex arithmetic")
+    end
+
```

Comment:
Does this mean that we cannot factorize small matrices using KrylovKit? It would have some additional complications for working with TensorKit, where the blocks could end up quite small.